File: framework/languages/java/e2e/SpotService/src/main/java/systems/zlink/e2e/spotservice/StateCommandHandler.java
package systems.zlink.e2e.spotservice;

import systems.zlink.framework.spots.ZLinkSpotPacketHandler;

public final class StateCommandHandler
    implements ZLinkSpotPacketHandler<UserSpot, Contracts.StateCommand> {
    @Override
    public void handle(
        UserSpot spot,
        Contracts.StateCommand message) {
        spot.command(message.value());
    }
}

File: framework/languages/java/e2e-kotlin/SpotService/Server/Play/src/main/kotlin/systems/zlink/e2e/kotlin/spotservice/play/handlers/StateCommandHandler.kt
package systems.zlink.e2e.kotlin.spotservice.play.handlers

import systems.zlink.e2e.kotlin.spotservice.Contracts
import systems.zlink.e2e.kotlin.spotservice.ScenarioState
import systems.zlink.e2e.kotlin.spotservice.play.spots.*
import systems.zlink.framework.spots.ZLinkSpotPacketHandler

class StateCommandHandler : ZLinkSpotPacketHandler<UserSpot, Contracts.StateCommand> {
    override fun handle(spot: UserSpot, message: Contracts.StateCommand) {
        spot.command(message.value)
    }
}
